metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -450,8 +452,48 @@ def _find_measure_recipe(
                 for y in x.join_on_partition_time_dimensions
             )
 
+            measure_node = node_with_lowest_cost
+            # Since every build_aggregate_measures function call groups measure_specs by additiveness
+            # all of these measure_specs will have the same non_additive_dimension_spec
+            non_additive_dimension_spec: Optional[NonAdditiveDimensionSpec] = measure_specs[
+                0
+            ].non_additive_dimension_spec
+            if non_additive_dimension_spec is not None:
+                # Gets the dimension given in the query that matches the non-additive dimension
+                primary_time_dimension = self._data_source_semantics.find_primary_time_dimension(
+                    self._data_source_semantics.get_data_sources_for_measure(measure_specs[0].as_reference)[0]
+                )
+                data_set = self._node_data_set_resolver.get_output_data_set(measure_node)
+                queried_time_dimension_spec: Optional[LinkableInstanceSpec] = None
+                for linkable_spec in linkable_specs:
+                    dimension_name_match = linkable_spec.element_name == non_additive_dimension_spec.name
+                    metric_time_match = (
+                        non_additive_dimension_spec.name == primary_time_dimension.element_name
```

Comment:
Does this work with custom measure agg time dimensions? The `find_primary_time_dimension` checks at the data source level, and I don't see anything looking at the measure itself, but the measure might well have a different time dimension set. Does the source node builder step deal with this for us? If so it'd be good to document this behavior, and I recommend doing it via a helper method that returns the queried_time_dimension_spec and can then be documented independently with its assumptions.

It's actually kind of unclear to me how all of these things fit together, honestly, so more clarity in this block would be super helpful.

Reply:
addressed here https://github.com/transform-data/metricflow/pull/253#discussion_r977898952